### PR TITLE
Added SesClient and tests

### PR DESCRIPTION
**Pull request by Moasib-Arif**
### What

Added python client for sending emails using aws ses and the boto3 client. 
And added tests and ses to the environment

### How to review

- Run all the test for test_email.py
- Check the test logic is correct
- check for the SesClient logic is correct
- Any typo in documentation or comments

### Who can review

Anyone

### CODE REVIEW

File: tests/email/test_email.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+@mock_aws
+def test_ses_client_send_missing_body(mock_ses_client):
+    """
+    Test the send method of the SesClient with a missing body.
+    """
+    # Initialise the SES client
+    mock_ses_client = SesClient('sender@example.com', 'us-west-2')
+
+    with pytest.raises(botocore.exceptions.ParamValidationError) as e:
```
Comment (SarahJohnsonONS) [inline tests/email/test_email.py:77]: Add an assertion for the content of the error message (e.g. `Invalid type for parameter Message.Body.Text.Data, value: None`)

Comment (SarahJohnsonONS) [inline tests/email/test_email.py:79]: Add an assertion for the content of the error message (e.g. `Invalid type for parameter Message.Body.Text.Data, value: None`)

File: tests/email/test_email.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+@mock_aws
+def test_ses_client_send_missing_subject(mock_ses_client):
+    """
+    Test the send method of the SesClient with a missing subject.
+    """
+    # Initialise the SES client
+    mock_ses_client = SesClient('sender@example.com', 'us-west-2')
+
+    with pytest.raises(botocore.exceptions.ParamValidationError) as e:
+        mock_ses_client.send('recipient@example.com', None, 'body')
```
Comment (SarahJohnsonONS) [inline tests/email/test_email.py]: Add an assertion for the content of the error message (e.g. `Invalid type for parameter Message.Subject.Data, value: None`)

**Review by SarahJohnsonONS — COMMENTED** [review]
Really nothing much I can see that needs changing - added a couple of comments, but otherwise good to go.

File: dpytools/email/ses/client.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+class SesClient:
+    """
+    A client for sending emails using AWS SES and the boto3 client.
+
+    Attributes:
+        client: The boto3 SES client.
+        sender: The sender's email address.
+    """
+
+    EMAIL_REGEX = r"(^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$)"
```
Comment (mikeAdamss) [inline dpytools/email/ses/client.py]: have a look at this: https://github.com/JoshData/python-email-validator , regex works but is fragile and really not that obvious. If someone wants to solve this for us, lets let them.

File: dpytools/email/ses/client.py [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+        Args:
+            sender (str): The sender's email address.
+            aws_region (str): The AWS region.
+
+        Raises:
+            ValueError: If the sender's email or AWS region is invalid.
+        """
+        # check sender is actually a valid email (as in the format)
+        if not re.match(self.EMAIL_REGEX, sender):
+            raise ValueError("Invalid sender email format")
```
Comment (mikeAdamss) [inline dpytools/email/ses/client.py]: show the thing that isn't valid please. Have a hard look at any error message your write without any kind of variable in it, it's very, very rare that you'll want a generic error message.

File: dpytools/email/ses/client.py [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+        
+        # check the aws region looks valid
+        if aws_region not in boto3.session.Session().get_available_regions('ses'):
+            raise ValueError("Invalid AWS region")
```
Comment (mikeAdamss) [inline dpytools/email/ses/client.py]: see other comment on generic error messages.

**Review by mikeAdamss — COMMENTED** [review]
Couple of minor comments only.

Can we add a readme for this as well please. We've done well at documenting this package of python helpers, lets keep it up.

**Review by mikeAdamss — COMMENTED** [review]
code looks good but you need to add the email_validator to poetry dependencies, cant run the tests at the moment

**Review by mikeAdamss — APPROVED** [review]
LGTM 👍

**Pull request merged**